### PR TITLE
Introduce golden trace concept [FND-44]

### PR DESCRIPTION
**Pull request by danmux**
This is the first introduction of the concept of a golden trace.

This is an addition to the API, and may be subject to change, as we observe how it behaves.

### CODE REVIEW

File: testing/jaeger/client.go [hunk 1]
```diff
@@ -73,10 +73,17 @@ func (j *Client) Traces(ctx context.Context, since time.Time) ([]Trace, error) {
 
 func AssertTag(t *testing.T, tags []Tag, k, v string) {
 	t.Helper()
+	if HasTag(tags, k, v) {
+		return
+	}
+	t.Errorf("key:%q with value %q was not found", k, v)
+}
+
+func HasTag(tags []Tag, k, v string) (bool, ) {
```
Comment (bjohnso5) [inline testing/jaeger/client.go:82]: `(bool, )` was this an artifact of refactoring? Or is this a Go idiom I've not seen before?

File: o11y/o11y.go [hunk 2]
```diff
@@ -117,6 +133,8 @@ type Span interface {
 	// End sets the duration of the span and tells the related provider that the span is complete,
 	// so it can do its appropriate processing. The span should not be used after End is called.
 	End()
+
+	//Link(Span)
```
Comment (bjohnso5) [inline o11y/o11y.go:137]: is this intended to be excluded from the interface? I see there's a note that it specifically won't be implemented for libhoney directly

File: testing/jaeger/client.go (hunk 1 above)
Comment (danmux) [reply testing/jaeger/client.go:82]: refactor fuckup

File: o11y/o11y.go (hunk 2 above)
Comment (danmux) [reply o11y/o11y.go:137]: ah - another refactor cleanup.

This is the problem of getting distracted on code that you started months ago :(

File: o11y/honeycomb/honeycomb.go [hunk 3]
```diff
@@ -26,6 +26,21 @@ type honeycomb struct {
 	metricsProvider o11y.ClosableMetricsProvider
 }
 
+func (h *honeycomb) StartGoldenTrace(ctx context.Context, name string) context.Context {
+	//TODO implement me
+	panic("implement me")
```
Comment (bjohnso5) [inline o11y/honeycomb/honeycomb.go:31]: is there any risk in trying to use this in a service that isn't OTel-enabled? I _think_ that you've already onboarded all the services, but I don't know if that's definitely true. If not, would it make more sense to just leave these as TODOs with a no-op return of the provided context?

**Review by bjohnso5 — APPROVED** [review]

File: o11y/honeycomb/honeycomb.go (hunk 3 above)
Comment (danmux) [reply o11y/honeycomb/honeycomb.go:31]: mebe - but if anyone is using these right now - i would want them to panic! - kinda even the ot1l ones - they are not ready. If anyone else uses them now - they have no idea what will happen :p

Comment (danmux) [reply o11y/honeycomb/honeycomb.go:31]: but yah - totes safer to remove the panic - good call :)

**Pull request merged**